core/arch/arm/plat-imx/imx_pl310.c

```diff
@@ -68,13 +68,6 @@ void arm_cl2_enable(vaddr_t pl310_base)
 
 vaddr_t pl310_base(void)
 {
-	static void *va;
-
-	if (cpu_mmu_enabled()) {
-		if (!va)
-			va = phys_to_virt(PL310_BASE, MEM_AREA_IO_SEC);
-		return (vaddr_t)va;
-	}
-	return PL310_BASE;
+	return core_mmu_get_io_va(PL310_BASE);
```

Comment:
@etienne-lms +1 with your statement: "`phys_to_virt_io()` api should be reserved to iomem for which the secure state does not really matter", that's exactly the purpose of the function (otherwise use `phys_to_virt()`), and in my mind it was the meaning of the "_io" suffix: it means "any of the `_IO_` memtyptes (secure or non-secure)".

So, for consistency with this naming scheme, we should have `core_mmu_get_io_va()` without a memtype parameter. And it's OK to use it here (`imx_pl310.c`) since as you said above the mem range is registered secure above.

Now, I don't mind also having `core_mmu_get_va(pa, type)` to complete the picture... but that can be added when needed.